### PR TITLE
chore: add 'system' context to `sp-theme`

### PR DESCRIPTION
**Pull request by rubencarvalho**
## Description

This PR adds a `systemContext` functionality to the `<sp-theme>` component, enabling descendant components to consume the system context in a reusable manner.

## Motivation and context

Added a`SystemResolutionController` to encapsulate the logic for requesting and receiving the system context.

Updated the `sp-theme` README with instructions on how to consume the system context using the controller.

## How has this been tested?

Integrated these changes in the `IconBase` on the S2 icon delivery to test its reactivity.
Unit tests WIP.

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/icon/src/Icon.ts [hunk 1]
```diff
@@ -36,13 +38,21 @@ export class Icon extends IconBase {
 
     private updateIconPromise?: Promise<void>;
 
+    private currentSystem: SystemVariant = '';
+    private unsubscribeSystemContext: (() => void) | null = null;
+
     public override connectedCallback(): void {
         super.connectedCallback();
```
Comment (Rajdeepc) [inline packages/icon/src/Icon.ts:45]: I feel a ghost call also takes up memory!

File: packages/icon/src/Icon.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@ import { IconsetRegistry } from '@spectrum-web-components/iconset/src/iconset-re
 
 import { IconBase } from './IconBase.js';
 
+type SystemVariant = 'spectrum' | 'spetrum-two' | 'express' | '';
```
Comment (Rajdeepc) [inline packages/icon/src/Icon.ts:24]:
```suggestion
import type {
    SystemVariant
} from '@spectrum-web-components/theme';
```

File: tools/theme/src/Theme.ts [hunk 3]
```diff
@@ -472,9 +306,48 @@ export class Theme extends HTMLElement implements ThemeKindProvider {
...
+
         this.updateComplete = this.__createDeferredPromise();
     }
 
+    private _systemContextConsumers = new Map<
+        HTMLElement,
+        [SystemContextCallback, () => void]
+    >();
+
+    private _handleSystemContext(
```
Comment (Rajdeepc) [inline tools/theme/src/Theme.ts:322]: Documentation block?

**Review by Rajdeepc — CHANGES_REQUESTED** [review]

**Review by Rajdeepc — COMMENTED** [review]
Neat! Also please add this controller in this [README.md] (https://github.com/adobe/spectrum-web-components/blob/testing-css-bridge-for-s2/tools/reactive-controllers/README.md#L0-L1) and we can land this!

**Review by Rajdeepc — APPROVED** [review]

**Comment by charlessuh** [comment]
@rubencarvalho: I believe `@spectrum-web-components/reactive-controllers` is [missing](https://github.com/adobe/spectrum-web-components/blob/8d2790376b9cc09b543dfde3dcb778709b4394f9/packages/icon/package.json#L73) as a dependency in the `package.json` of the icon package. Can you add it?

**Comment by TarunAdobe** [comment]
@charlessuh this `reactive-controller` is not meant to be used outside of our library and we haven't documented it either. Could you please explain better how this missing dependency is affecting your use-case?

**Pull request merged**